lib/utils.js

```diff
@@ -0,0 +1,37 @@
+const { curry } = require('ramda');
+const { KEY_START, KEY_END, RSA_KEY_START, RSA_KEY_END } = require('./constants');
+const domainRegTest = /^[a-z0-9-]+$/;
+
+const tryParseJson = (keyjson) => {
+  if (typeof keyjson === 'string') {
+    try {
+      keyjson = JSON.parse(keyjson);
+    } catch (e) {
+      throw Error(`Could not parse json properly: ${e.message}`);
+    }
+  }
+  return keyjson;
+};
+
+const isSslKey = (key) => {
+  return !(key.startsWith(KEY_START) && key.endsWith(KEY_END)) && !(key.startsWith(RSA_KEY_START) && key.endsWith(RSA_KEY_END))
+};
+
+const isDomainMatch = curry((toCheck, base) => {
+  if (base === toCheck) return true;
+  
+  if (!base.startsWith('*')) return false; // exact domain and already does not match
+  
+  if (!toCheck.endsWith(base)) return false; // it's not a subdomain
+
+  const restOfDomain = toCheck.slice(0, -base.length);
```

Comment:
It doesn't look like we have any strict guarantees that `base` is a String from the caller `validateCertToDomain`. AFAICT, the `domain` is being passed in by the user. I think we should handle them giving us garbage and/or unexpected types. That will save some bad method checks in this method that could throw unhandled exceptions.